server/src-lib/Hasura/GraphQL/Schema/Insert.hs

```diff
@@ -0,0 +1,71 @@
+module Hasura.GraphQL.Schema.Insert where
+
+import           Hasura.Prelude
+
+import qualified Hasura.RQL.DML.Insert.Types    as RQL
+import qualified Hasura.RQL.DML.Returning.Types as RQL
+
+import           Hasura.RQL.Types.BoolExp
+import           Hasura.RQL.Types.Column
+import           Hasura.RQL.Types.Common
+import           Hasura.SQL.Types
+
+
+-- WIP NOTE
+--
+-- An abstract representation of insert was tricky to pin down. The
+-- following structures are taken from Resolve, and slightly
+-- modified. What needs to be decided is where those should go and
+-- what we should do with them, really.
+--
+-- They are currently in this file in order to be accessible from the
+-- Context file without creating a dependency loop, but this is only a
+-- temporary emplacement: they are not part of the Schema and don't
+-- belong in this folder.
```

Comment:
What's to be done about this note?

Reply:
I don't know, to be honest. On one hand: while this representation isn't perfect, it works. But the question of where it should be put is still open: for every other kind of mutation / query, we've been using RQL structures as our internal representation, inserts are the only exception; I don't think it belongs here alongside the parsers. Ultimately, it should be integrated with the rest of RQL, especially if we deprecate RQL from an actual query language to just an internal intermediate representation.

For now, I think we can replace this long comment by two FIXMEs:

- one about moving those data types to RQL
- one about simplifying them since "we" (the new schema code) are the only users of it.

Reply:
I pushed a better comment.